site/src/pages/TemplateVariablesPage/TemplateVariablesPageView.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  CreateTemplateVersionRequest,
+  TemplateVersion,
+  TemplateVersionVariable,
+} from "api/typesGenerated"
+import { AlertBanner } from "components/AlertBanner/AlertBanner"
+import { Loader } from "components/Loader/Loader"
+import { ComponentProps, FC } from "react"
+import { TemplateVariablesForm } from "./TemplateVariablesForm"
+import { Stack } from "components/Stack/Stack"
+import { makeStyles } from "@material-ui/core/styles"
+import { useTranslation } from "react-i18next"
+import { FullPageHorizontalForm } from "components/FullPageForm/FullPageHorizontalForm"
+
+export interface TemplateVariablesPageViewProps {
+  templateVersion?: TemplateVersion
+  templateVariables?: TemplateVersionVariable[]
+  onSubmit: (data: CreateTemplateVersionRequest) => void
+  onCancel: () => void
+  isSubmitting: boolean
+  errors?: {
+    getTemplateError?: unknown
+    getActiveTemplateVersionError?: unknown
+    getTemplateVariablesError?: unknown
+    updateTemplateError?: unknown
+  }
+  initialTouched?: ComponentProps<
+    typeof TemplateVariablesForm
+  >["initialTouched"]
+}
+
+export const TemplateVariablesPageView: FC<TemplateVariablesPageViewProps> = ({
+  templateVersion,
+  templateVariables,
+  onCancel,
+  onSubmit,
+  isSubmitting,
+  errors = {},
+  initialTouched,
+}) => {
+  const classes = useStyles()
+  const isLoading =
+    !templateVersion &&
+    !templateVariables &&
+    !errors.getTemplateError &&
+    !errors.getTemplateVariablesError &&
+    !errors.updateTemplateError
+  const { t } = useTranslation("templateVariablesPage")
+
+  return (
+    <FullPageHorizontalForm title={t("title")} onCancel={onCancel}>
+      {Boolean(errors.getTemplateError) && (
+        <Stack className={classes.errorContainer}>
+          <AlertBanner severity="error" error={errors.getTemplateError} />
+        </Stack>
+      )}
+      {Boolean(errors.getActiveTemplateVersionError) && (
+        <Stack className={classes.errorContainer}>
+          <AlertBanner
+            severity="error"
+            error={errors.getActiveTemplateVersionError}
+          />
+        </Stack>
+      )}
+      {Boolean(errors.getTemplateVariablesError) && (
+        <Stack className={classes.errorContainer}>
+          <AlertBanner
+            severity="error"
+            error={errors.getTemplateVariablesError}
+          />
+        </Stack>
+      )}
+      {Boolean(errors.updateTemplateError) && (
+        <Stack className={classes.errorContainer}>
+          <AlertBanner severity="error" error={errors.updateTemplateError} />
+        </Stack>
+      )}
```

Comment:
Idk if we want to be very specific on errors since the API already gives us the scoped error message. I would condense all the errors into one or two - one for the getting data and another for the submit action. This is a pattern we used in the past and adds more complexity than value.

Reply:
Fixed. I will just use `getTemplateDataError` and `updateTemplateError`.